### PR TITLE
feat_community_chat_screen

### PR DESCRIPTION
**Pull request by Paulmburu**
Screenshot_1633422192
Screenshot_1633438624

### CODE REVIEW

File: lib/presentation/community/chat_screen/widgets/awaiting_moderation_widget.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/// [AwaitingModerationWidget] used within [SentMessageItem] to show messages awaiting moderation.
+class AwaitingModerationWidget extends StatelessWidget {
+  /// [AwaitingModerationWidget] used within [SentMessageItem] to show messages awaiting moderation.
+  ///
+  const AwaitingModerationWidget();
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/awaiting_moderation_widget.dart:15]: is this `GestureDetector` necessary?

File: lib/presentation/community/chat_screen/widgets/bottom_text_box.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:myafyahub/domain/core/value_objects/app_strings.dart';
+import 'package:myafyahub/domain/core/value_objects/asset_strings.dart';
+import 'package:myafyahub/presentation/core/theme/theme.dart';
+import 'package:shared_themes/spaces.dart';
+
+class BottomTextBox extends StatelessWidget {
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/bottom_text_box.dart:8]: change the naming of this to match the contents specified in the documentation. for example `ChatInput` or `MessageInput`

File: lib/presentation/community/chat_screen/widgets/bottom_text_box.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                  padding: const EdgeInsets.all(10),
+                  child: SvgPicture.asset(
+                    attachFileIcon,
+                    width: 18,
+                    height: 18,
+                  ),
+                ),
+              ),
+              smallHorizontalSizedBox,
+              GestureDetector(
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/bottom_text_box.dart:53]: attach these `GestureDetectors` to a coming soon `SnackBar`

File: lib/presentation/community/chat_screen/widgets/highlighted_message_widget.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+import 'package:myafyahub/presentation/core/theme/theme.dart';
+import 'package:shared_themes/spaces.dart';
+import 'package:shared_themes/text_themes.dart';
+
+class HighlightedMessageWidget extends StatelessWidget {
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/highlighted_message_widget.dart:6]: same to this. change the name to `QuotedMessageWidget`

File: lib/presentation/community/chat_screen/widgets/receive_message_item.dart [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import 'package:misc_utilities/number_constants.dart';
+import 'package:myafyahub/presentation/community/chat_screen/widgets/sender_type_widget.dart';
+import 'package:myafyahub/presentation/core/theme/theme.dart';
+import 'package:shared_themes/spaces.dart';
+import 'package:shared_themes/text_themes.dart';
+
+import 'highlighted_message_widget.dart';
+
+/// [ReceiveMessageItem] used within the community chat screen to show received messages.
+class ReceiveMessageItem extends StatelessWidget {
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/receive_message_item.dart:11]: this should be `ReceivedMessageItem`. remember to also change the file names

File: lib/presentation/community/chat_screen/widgets/send_message_item.dart [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import 'package:misc_utilities/number_constants.dart';
+import 'package:myafyahub/presentation/community/chat_screen/widgets/awaiting_moderation_widget.dart';
+import 'package:myafyahub/presentation/core/theme/theme.dart';
+import 'package:shared_themes/spaces.dart';
+import 'package:shared_themes/text_themes.dart';
+
+import 'highlighted_message_widget.dart';
+
+/// [SendMessageItem] used within the community chat screen to show sent messages.
+class SendMessageItem extends StatelessWidget {
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/send_message_item.dart:11]: change naming as well to `SentMessageItem`

File: lib/presentation/community/chat_screen/widgets/sender_type_widget.dart [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    required this.senderTitle,
+    required this.senderIcon,
+  });
+
+  final String senderTitle;
+  final IconData senderIcon;
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
```
Comment (abiudrn) [inline lib/presentation/community/chat_screen/widgets/sender_type_widget.dart:22]: remove these `GestureDetectors` if they are not being used at the moment

**Review by abiudrn — CHANGES_REQUESTED** [review]

File: lib/presentation/community/chat_screen/widgets/awaiting_moderation_widget.dart (hunk 1 above)
Comment (Paulmburu) [reply lib/presentation/community/chat_screen/widgets/awaiting_moderation_widget.dart:17]: No, let me get rid of it

**Review by abiudrn — APPROVED** [review]

**Pull request merged**